src/she/sdl2/she.cpp

```diff
@@ -0,0 +1,521 @@
+        static SDL2EventQueue g_queue;
+        return &g_queue;
+    }
+
+    class SDL2System : public CommonSystem {
+    public:
+        SDL2System() {
+            g_instance = this;
+        }
+
+        ~SDL2System() {
+            IMG_Quit();
+            SDL_Quit();
+            g_instance = nullptr;
+        }
+
+        void dispose() override {
+            delete this;
+        }
+
+        void activateApp() override {
+            // Do nothing
+        }
+
+        void finishLaunching() override {
+            // Do nothing
+        }
+
+        Capabilities capabilities() const override {
+            return (Capabilities)(Capabilities::CanResizeDisplay);
+        }
+
+        EventQueue* eventQueue() override { // TODO remove this function
+            return EventQueue::instance();
+        }
+
+        bool gpuAcceleration() const override {
+            return true;
+        }
+
+        void setGpuAcceleration(bool state) override {
+            // Do nothing
+        }
+
+        gfx::Size defaultNewDisplaySize() override {
+            return gfx::Size(0, 0);
+        }
+
+        Display* defaultDisplay() override {
+            return unique_display;
+        }
+
+        Display* createDisplay(int width, int height, int scale) override {
+            //LOG("Creating display %dx%d (scale = %d)\n", width, height, scale);
+            return new SDL2Display(width, height, scale);
+        }
+
+        Surface* createSurface(int width, int height) override {
+            return new SDL2Surface(width, height, SDL2Surface::DeleteAndDestroy);
+        }
+
+        Surface* createRgbaSurface(int width, int height) override {
+            return new SDL2Surface(width, height, 32, SDL2Surface::DeleteAndDestroy);
+        }
+
+        Surface* loadSurface(const char* filename) override {
+            SDL_Surface* bmp = IMG_Load(filename);
+            if (!bmp)
+              throw std::runtime_error("Error loading image");
+
+            std::cout << "Loading " << filename << " "
+                      << std::to_string(bmp->format->BitsPerPixel) << " "
+                      << std::to_string(bmp->format->Rshift) << " "
+                      << std::to_string(bmp->format->Ashift) << std::endl;
+
+            return new SDL2Surface(bmp, SDL2Surface::DeleteAndDestroy);
+        }
+
+        Surface* loadRgbaSurface(const char* filename) override {
+            return loadSurface(filename);
+        }
+
+    };
+
+    System* create_system() {
+        return new SDL2System();
+    }
+
+    System* instance()
+    {
+        return g_instance;
+    }
+
+    void error_message(const char* msg)
+    {
+        if (g_instance && g_instance->logger())
+            g_instance->logger()->logError(msg);
+
+#ifdef _WIN32
+        std::wstring wmsg = base::from_utf8(msg);
+        std::wstring title = base::from_utf8(PACKAGE);
+        ::MessageBoxW(NULL, wmsg.c_str(), title.c_str(), MB_OK | MB_ICONERROR);
```

Comment:
Is there a cross-platform way to do this?

Reply:
`SDL_ShowSimpleMessageBox`, maybe?

Reply:
Looks good.